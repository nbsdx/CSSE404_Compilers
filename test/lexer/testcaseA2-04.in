	public static String getGCD(String a, String b) {
		System.out.println((+a+,+b+));
		if(b.equals(0))
			return a;
		String temp = StringMath.mod(a,b);
		a = b;
		b = temp;
		return getGCD(b,temp);
	}
	public static void main(String args[]) {
		int prod = 52*60;
		int e = 17;
		//System.out.println(getGCD(prod,e));
		//System.out.println(getGCD(prod+,e+));
		System.out.println(4626);
	}
}