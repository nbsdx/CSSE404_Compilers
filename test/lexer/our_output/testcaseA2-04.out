ReservedWord, public
ReservedWord, static
ReservedWord, String
ID, getGCD
Delimiter, (
ReservedWord, String
ID, a
Delimiter, ,
ReservedWord, String
ID, b
Delimiter, )
Delimiter, {
ReservedWord, System.out.println
Delimiter, (
Delimiter, (
Operator, +
ID, a
Operator, +
Delimiter, ,
Operator, +
ID, b
Operator, +
Delimiter, )
Delimiter, )
Delimiter, ;
ReservedWord, if
Delimiter, (
ID, b
Delimiter, .
ID, equals
Delimiter, (
Integer, 0
Delimiter, )
Delimiter, )
ReservedWord, return
ID, a
Delimiter, ;
ReservedWord, String
ID, temp
Delimiter, =
ID, StringMath
Delimiter, .
ID, mod
Delimiter, (
ID, a
Delimiter, ,
ID, b
Delimiter, )
Delimiter, ;
ID, a
Delimiter, =
ID, b
Delimiter, ;
ID, b
Delimiter, =
ID, temp
Delimiter, ;
ReservedWord, return
ID, getGCD
Delimiter, (
ID, b
Delimiter, ,
ID, temp
Delimiter, )
Delimiter, ;
Delimiter, }
ReservedWord, public
ReservedWord, static
ReservedWord, void
ReservedWord, main
Delimiter, (
ReservedWord, String
ID, args
Delimiter, [
Delimiter, ]
Delimiter, )
Delimiter, {
ReservedWord, int
ID, prod
Delimiter, =
Integer, 52
Operator, *
Integer, 60
Delimiter, ;
ReservedWord, int
ID, e
Delimiter, =
Integer, 17
Delimiter, ;
ReservedWord, System.out.println
Delimiter, (
Integer, 4626
Delimiter, )
Delimiter, ;
Delimiter, }
Delimiter, }
